tests/std/include/test_atomic_wait.hpp

```diff
@@ -174,6 +175,10 @@ inline void test_atomic_wait() {
     test_atomic_wait_func(three_chars{1, 1, 3}, three_chars{1, 2, 3}, waiting_duration);
     test_atomic_wait_func(big_char_like{'a'}, big_char_like{'b'}, waiting_duration);
 
+    test_atomic_wait_func(std::make_shared<int>('a'), std::make_shared<int>('b'), waiting_duration);
+    test_atomic_wait_func(
+        std::weak_ptr(std::make_shared<int>('a')), std::weak_ptr(std::make_shared<int>('b')), waiting_duration);

```

Comment:
Style request: we're increasingly using braces for temporary objects (not universally for 1+ args, but I think there's a minor readability advantage). Also you're already using braces above.
```suggestion
        std::weak_ptr{std::make_shared<int>('a')}, std::weak_ptr{std::make_shared<int>('b')}, waiting_duration);
```